Allow setting the mode of log files

andrew-d opened the pull request:
This is useful in cases where you want an external process (e.g. a log forwarder running with dropped privs) to be able to read the log files, but still want to run `osqueryd` as root in order to gather information.

Quick note: I'm not sure why, but the test isn't working as expected.  Any idea why that might be?  This is the only actual mode I was able to find in the codebase.

osqueryer commented:
Can one of the osquery admins reply with "ok to test" to kick off a build...

theopolis commented on osquery/logger/plugins/filesystem.cpp:
```diff
@@ -24,9 +25,15 @@ FLAG(string,
 /// Legacy, backward compatible "osquery_log_dir" CLI option.
 FLAG_ALIAS(std::string, osquery_log_dir, logger_path);
 
+FLAG(string,
```
You can make this an `int32` and avoid the string->int conversion.

theopolis commented on tools/tests/test_osqueryd.py:
```diff
@@ -93,5 +93,38 @@ def test_5_daemon_sigint(self):
         self.assertTrue(daemon.retcode in acceptable_retcodes)
 
+    def test_6_logger_mode(self):
+        logger_path = os.path.join(test_base.CONFIG_DIR, "logger-mode-tests")
```
Right! So this is testing the status logs: INFO, WARNING, ERROR emitted by glog. These are not the same as the results logging which are the added/removed/etc to a default `./osqueryd.results.log` which is what the `FilesystemLoggerPlugin` is writing within `logString`. To get the result log to produce some data we could do either:

A) Have the `FilesystemLoggerPlugin` create the intended log file without any bytes on startup/construction. 
B) Add a small config to this integration test that scheduled a "select * from time" with an interval=1 and wait until the log is created.

A takes a bit more code, B is a little janky will bloat the python tests by at least another 1s, but is easy and fast!

theopolis commented:
To get both the results logs and the [glogging status logs](https://github.com/google/glog/blob/master/src/logging.cc#L904) to use the same mode you'll have to add:
```cpp
DECLARE_int32(logfile_mode)
```

Then in the `FilesystemLoggerPlugin::setUp` method:
```cpp
FLAGS_logfile_mode = FLAGS_logger_mode;
```

andrew-d commented:
So, it looks like the version of `glog` in `third_party` doesn't include google/glog#26.  I'm going to bump the dependency locally and try that out.

andrew-d commented:
Okay, I updated everything and it works locally.  I've also sent osquery/third-party#10 - after that is merged, and the submodule dependency here is bumped, this should be good to go!

theopolis commented:
Merged! You'll need to update the submodule commit hash to the new merge commit in third-party. Feel free to just `git commit --amend` and `git push -f` that commit update to this PR.

andrew-d commented:
Okay, updated!  I also squashed into a single commit.  If you wouldn't mind running tests on the buildbot (just to be safe...), that would be greatly appreciated :smile:

theopolis commented:
ok to test

osqueryer commented:
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:
